### PR TITLE
Update dependencies

### LINKED ISSUE #313: Upgrade node-gyp for Python 3.12 support
See also https://github.com/nodejs/node/pull/50209

GitHub Actions will switch to that Python version by default sooner or later.

### PR DESCRIPTION
**Pull request by orgads**
Excluding glob and standard, which require code changes.

### CODE REVIEW

**Comment by orgads** [comment]
These were 2 separate commits - one for standard (which required all the const changes) and one for the rest.

I rebased and swapped them.

File: package.json [hunk 1]
```diff
@@ -21,16 +21,16 @@
...
   "dependencies": {
-    "cmake-js": "^7.2.1",
+    "cmake-js": "^7.3.0",
     "detect-libc": "^2.0.2",
     "each-series-async": "^1.0.1",
     "execspawn": "^1.0.1",
     "ghreleases": "^3.0.2",
     "github-from-package": "0.0.0",
-    "glob": "^7.2.3",
+    "glob": "^10.3.10",
```
Comment (lovell) [inline package.json:30]: `glob` versions 9+ use a Promise-based API so this upgrade will require a code change (and highlights a possible unit test blind spot).
https://github.com/prebuild/prebuild/blob/f8d82a1d9f1ae17e2a389414d242e22ee8d2efde/bin.js#L45

File: package.json [hunk 2]
```diff
@@ -39,13 +39,13 @@
     "nw-gyp": "^3.6.6",
     "rc": "^1.2.8",
     "run-waterfall": "^1.1.7",
-    "tar-stream": "^3.1.6"
+    "tar-stream": "^3.1.7"
   },
   "devDependencies": {
     "a-native-module": "^1.0.0",
-    "rimraf": "^4.4.1",
+    "rimraf": "^5.0.5",
```
Comment (lovell) [inline package.json:46]: Can we remove this dependency entirely by switching to `fs.rmSync(path, { recursive: true })`?

**Review by lovell — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (orgads) [reply package.json]: Thanks for spotting that! I moved it to a separate PR (depends on this one though).

I don't know how to test it without actually uploading stuff :)

File: package.json (hunk 2 above)
Comment (orgads) [reply package.json]: It doesn't work well. See #319

**Review by lovell — APPROVED** [review]

**Pull request merged**